Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -889,22 +889,33 @@ private final class TokenStreamCreator: SyntaxVisitor {
   }
 
   override func visit(_ node: ClosureSignatureSyntax) -> SyntaxVisitorContinueKind {
-    let consistency: GroupBreakStyle
-    if let input = node.input, input.is(ClosureParamListSyntax.self) {
-      consistency = argumentListConsistency()
-    } else {
-      consistency = .inconsistent
-    }
-
     before(node.firstToken, tokens: .open)
 
     if let input = node.input {
       // We unconditionally put a break before the `in` keyword below, so we should only put a break
       // after the capture list's right bracket if there are arguments following it or we'll end up
       // with an extra space if the line doesn't wrap.
       after(node.capture?.rightSquare, tokens: .break(.same))
-      before(input.firstToken, tokens: .open(consistency))
-      after(input.lastToken, tokens: .close)
+
+      // When it's parenthesized, the input is a `ParameterClauseSyntax`. Otherwise, it's a
+      // `ClosureParamListSyntax`. The parenthesized version is wrapped in open/close breaks so that
+      // the parens create an extra level of indentation.
+      if let parameterClause = input.as(ParameterClauseSyntax.self) {
```

Comment:
I just realized there are a couple things we need to be careful about here. When we added the `prioritizeKeepingFunctionOutputTogether` setting, we had to split where the open/close tokens were added due to visitation order. So if a `ParameterClause` can occur here and that setting is enabled, we won't compensate for that here.

Can you add a few tests with `prioritizeKeepingFunctionOutputTogether` that cover various combinations of signatures with throws/return types and fix up what needs to be handled for that? A test or two with an empty argument list (`{ () in ... }`) would be good too for completeness.

I don't know if we care about changing the behavior in the closure case for that setting. I guess we _could_ choose between this (`prioritizeKeepingFunctionOutputTogether = false`):

```swift
let x = {
  (foo: Int, bar: Int) throws
    -> Int in
  statements
}
```

and this (`prioritizeKeepingFunctionOutputTogether = true`):

```swift
let x = {
  (
    foo: Int,
    bar: Int
  ) throws -> Int in
  statements
}
```

Or we could always go with the second one. I'm not sure off the top of my head what we do today.

Reply:
So it turns that today using `prioritizeKeepingFunctionOutputTogether = true` crashes for closures with parenthesized parameter lists (i.e. `ParameterClauseSyntax`). Since it is very "function-like", I think respecting that configuration makes sense.

I had to rework the groups around the parameter clause, since the group around the output can't coexist with a group around the parens.